homeassistant/components/media_player/gpmdp.py

```diff
@@ -6,42 +6,144 @@
 """
 import logging
 import json
+import os
 import socket
 
 from homeassistant.components.media_player import (
     MEDIA_TYPE_MUSIC, SUPPORT_NEXT_TRACK, SUPPORT_PREVIOUS_TRACK,
     SUPPORT_PAUSE, MediaPlayerDevice)
 from homeassistant.const import (
     STATE_PLAYING, STATE_PAUSED, STATE_OFF)
+from homeassistant.loader import get_component
 
 _LOGGER = logging.getLogger(__name__)
 REQUIREMENTS = ['websocket-client==0.37.0']
 SUPPORT_GPMDP = SUPPORT_PAUSE | SUPPORT_PREVIOUS_TRACK | SUPPORT_NEXT_TRACK
+GPMDP_CONFIG_FILE = 'gpmpd.conf'
+_CONFIGURING = {}
 
 
-def setup_platform(hass, config, add_devices, discovery_info=None):
-    """Setup the GPMDP platform."""
-    from websocket import create_connection
+def request_configuration(hass, config, websocket, add_devices_callback):
+    """Request configuration steps from the user."""
+    configurator = get_component('configurator')
+    if 'gpmdp' in _CONFIGURING:
+        configurator.notify_errors(
+            _CONFIGURING['gpmdp'], "Failed to register, please try again.")
+
+        return
+
+    # pylint: disable=unused-argument
+    def gpmdp_configuration_callback(callback_data):
+        """The actions to do when our configuration callback is called."""
+        while True:
+            msg = json.loads(websocket.recv())
+            if msg['channel'] == 'connect':
+                if msg['payload'] == "CODE_REQUIRED":
+                    websocket.send('{"namespace": "connect",'
+                                   '"method": "connect",'
+                                   '"arguments": ["Home Assistant",'
+                                   ' "' + callback_data.get('pin') + '"]}')
```

Comment:
It seems like you're trying to send JSON here, why not use `json.dumps({ 'namespace': 'connect', … })`